velox/dwio/dwrf/writer/ColumnWriter.cpp

```diff
@@ -697,6 +697,89 @@ class TimestampColumnWriter : public BaseColumnWriter {
   std::unique_ptr<IntEncoder<false>> nanos_;
 };
 
+class DecimalColumnWriter : public BaseColumnWriter {
+ public:
+  DecimalColumnWriter(
+      WriterContext& context,
+      const TypeWithId& type,
+      uint32_t sequence,
+      std::function<void(IndexBuilder&)> onRecordPosition)
+      : BaseColumnWriter{context, type, sequence, onRecordPosition},
+        type_{type.type()},
+        unscaledValues_{createDirectEncoder</* isSigned = */ true>(
+            newStream(StreamKind::StreamKind_DATA),
+            getConfig(Config::USE_VINTS),
+            type_->isShortDecimal() ? LONG_BYTE_SIZE : 2 * LONG_BYTE_SIZE)},
+        scales_{createRleEncoder</* isSigned = */ true>(
+            RleVersion_1,
+            // DWRF's NANO_DATA has the same enum value as ORC's SECONDARY.
+            newStream(StreamKind::StreamKind_NANO_DATA),
+            context.getConfig(Config::USE_VINTS),
+            LONG_BYTE_SIZE)} {
+    reset();
+  }
+
+  uint64_t write(const VectorPtr& slice, const common::Ranges& ranges)
+      override {
+    VELOX_CHECK(
+        slice->type()->equivalent(*type_),
+        "Unexpected vector type: {}.",
+        slice->type()->toString());
+    auto localDecoded = decode(slice, ranges);
+    auto& decodedVector = localDecoded.get();
+    writeNulls(decodedVector, ranges);
+
+    size_t count = 0;
+    auto [_, scale] = getDecimalPrecisionScale(*type_);
+
+    for (auto& pos : ranges) {
+      if (!decodedVector.mayHaveNulls() || !decodedVector.isNullAt(pos)) {
+        if (type_->isShortDecimal()) {
```

Comment:
and type condition

Reply:
Added separate branch according to `isShortDecimal`. Thanks.